Associations

abhikanojia opened the pull request.

Commit:
Started Associations.. added association to cart and product models.

Commit:
Completed upto 5th point of depot extension.. yet to be implemented

abhishekkanojia committed:
Completed Associations

abhishekkanojia committed:
Added end next line to controller concern user_logged_in

harsh-ag commented on depot/Gemfile:
```diff
@@ -32,6 +32,8 @@ gem 'jquery-ui-rails'
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
 gem 'jbuilder', '~> 2.5'
+# pagination gem
+gem 'kaminari', :git => "git://github.com/amatsuda/kaminari.git", :branch => 'master'
```
lock gem version

harsh-ag commented on depot/app/models/cart.rb:
```diff
@@ -1,5 +1,7 @@
-  has_many :line_items, dependent: :destroy
+  has_many :line_items
+  has_many :products, through: :line_items
+  has_many :enabled_products, -> { where(enabled: true) }, through: :line_items, source: :product
```
create a scope for where(enabled: true) in product.rb

harsh-ag commented on depot/app/models/product.rb:
```diff
@@ -3,8 +3,9 @@ class Product < ApplicationRecord
   DEFAULT_TITLE = 'abc'
 
-  has_many :line_items
+  has_many :line_items, dependent: :restrict_with_error
```
comment  ensure_not_referenced_by_line_item callback

harsh-ag commented on depot/app/models/line_item.rb:
```diff
@@ -1,6 +1,6 @@
 class LineItem < ApplicationRecord
   belongs_to :product
-  belongs_to :cart
+  belongs_to :cart, counter_cache: true
```
where is migration for line_items_count

harsh-ag commented on depot/app/models/user.rb:
```diff
@@ -8,10 +8,17 @@ class User < ApplicationRecord
   before_update :ensure_not_updating_admin
   after_create_commit :send_welcome_email
+  # associations
+  has_many :orders
```
add dependent

harsh-ag commented on depot/app/models/order.rb:
```diff
@@ -14,8 +15,15 @@ class Order < ApplicationRecord
   end
+
+  def gross_total
+    total_amount = 0
```
can be written in one line

harsh-ag commented on depot/app/controllers/users_controller.rb:
```diff
@@ -68,6 +68,14 @@ def destroy
     end
   end
 
+  def show_orders
```
rename to orders

harsh-ag commented on depot/config/routes.rb:
```diff
@@ -7,6 +7,9 @@
     delete 'logout' => :destroy
   end
 
+  get '/users/orders' => 'users#show_orders'
```
use resources to create routes

harsh-ag commented on depot/app/controllers/orders_controller.rb:
```diff
@@ -28,7 +28,7 @@ def edit
   # POST /orders.json
   def create
-    @order = Order.new(order_params)
+    @order = current_user.create_new_order(order_params)
```
why have you done this.?

Commit:
Fixed gem version for kaminari.
Added dependent to orders model.
Refactored code for line_items gross_total function.
Added new line at end of show_orders.html.erb

Commit:
Removed create_new_order from user model.
changed order_controller accordingly.

harsh-ag commented on depot/Gemfile:
```diff
@@ -32,6 +32,8 @@ gem 'jquery-ui-rails'
 # Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
 gem 'jbuilder', '~> 2.5'
+# pagination gem
+gem 'kaminari', '~> 1.1.1', :git => "git://github.com/amatsuda/kaminari.git", :branch => 'master'
```
no need to mention git and branch

harsh-ag commented on depot/app/controllers/users_controller.rb:
```diff
@@ -68,6 +68,14 @@ def destroy
+    @orders = current_user.orders
+  end
+
+  def show_line_items
```
rename to line_items

harsh-ag commented on depot/app/views/users/show_line_items.html.erb:
```diff
@@ -0,0 +1,24 @@
+  <%= page_entries_info @line_items %>
+<% else %>
+  <h3>No Line Items found.</h3>
+<% end %>
```
leave last line

Commit:
Changed action name from show_line_items => line_items, show_orders => orders
removed git link and branch from gem file.
changed routes and rename files in users view.
removed duplicate files.
